merino/curated_recommendations/provider.py

```diff
@@ -107,39 +111,27 @@ def derive_region(locale: Locale, region: str | None = None) -> str | None:
         else:
             return None
 
-    async def fetch(
-        self, curated_recommendations_request: CuratedRecommendationsRequest
-    ) -> CuratedRecommendationsResponse:  # noqa
-        """Provide curated recommendations."""
-        # Get the recommendation surface ID based on passed locale & region
-        surface_id = CuratedRecommendationsProvider.get_recommendation_surface_id(
-            curated_recommendations_request.locale, curated_recommendations_request.region
-        )
-
-        corpus_items = await self.corpus_backend.fetch(surface_id)
-
-        # Convert the CorpusItem list to a CuratedRecommendation list.
-        recommendations = [
-            CuratedRecommendation(
-                **item.model_dump(),
-                receivedRank=rank,
-            )
-            for rank, item in enumerate(corpus_items)
-        ]
-
+    def process_recommendations(
```

Comment:
Would this name be more descriptive?
```suggestion
    def rank_main_recommendations(
```

Reply:
agree with mathijs here - i like `rank_recommendations` (as that makes it more general purpose should we end up ranking future feeds).